apps/aot/src/app/events/[year]/leaderboard/_components/daily-leaderboard.tsx

```diff
@@ -24,21 +27,21 @@ const getFirst100SubmissionsRanked = async (adventYear: string, adventDay: strin
     },
     take: LEADERBOARD_RANKING_LIMIT,
   });
-  return submissions;
+  const submissionsWithFlatUsers = submissions.map((s) => ({
+    //Named score since this shape of this type is being shared with getOverallLeaderboard
+    score: calculateDuration({ year: adventYear, day: adventDay }, s.createdAt),
+    //todo
+    isSupporter: Math.random() > 0.8,
+    ...s.user,
+  }));
+  return submissionsWithFlatUsers;
 };
 
-export default async function DailyLeaderboard({
-  adventYear,
-  adventDay,
-}: {
-  adventYear: string;
-  adventDay: string;
-}) {
-  const first100SubmissionsRanked = await getFirst100SubmissionsRanked(adventYear, adventDay);
-
-  return (
-    <div className="p-4">
-      <DataTableLeaderboard data={first100SubmissionsRanked} columns={dailyLeaderboardColumns} />
-    </div>
+const calculateDuration = (start: { year: string; day: string }, end: Date) => {
+  const duration = differenceInMilliseconds(
+    end,
+    new Date(Number(start.year), 11, Number(start.day)),
   );
-}
+  const date = format(duration, 'HH:MM:SS');
+  return date;
+};
```

Comment:
Not sure I fully understand the purpose of this function! Are you able to elaborate?

Reply:
This is to show the time taken to solve the challenge. The database has '2024-01-25T06:07:44.533Z' as created at, and I want to show how many HH:MM:SS has it elapsed since the start of the challenge day, to get to this time.

Reply:
Feel free to use `formatDate` function here: 

https://github.com/typehero/typehero/blob/94fe390088b79949ffa089f35387486fb0dd06e0/apps/aot/src/app/events/%5Byear%5D/leaderboard/_components/columns.tsx#L74